### PR TITLE
add deprecated section rpk

### PR DESCRIPTION
**Pull request by Deflaimun**
Resolves https://github.com/redpanda-data/documentation-private/issues/815

Preview: 
https://deploy-preview-265--redpanda-docs-preview.netlify.app/current/upgrade/deprecated/

### CODE REVIEW

File: modules/upgrade/pages/deprecated/index.adoc [hunk 1]
```diff
@@ -14,6 +14,25 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+| 23.2.1
+| Use the xref:23.2@upgrade:deprecated/cluster-resource.adoc[Redpanda resource] instead.
+
+|===
+
+## rpk commands
+
+All rpk commands that got modified or had flag changes.
+
+### Rename
```
Comment (Deflaimun) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
### Renames
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 2]
```diff
@@ -14,6 +14,25 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+
+| xref:23.2@reference:redpanda-operator/index.adoc[Cluster and Console custom resources]
+| 23.2.1
+| Use the xref:23.2@upgrade:deprecated/cluster-resource.adoc[Redpanda resource] instead.
+
+|===
+
+## rpk commands
+
+All rpk commands that got modified or had flag changes.
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
All rpk commands that have been modified or have had flag updates.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 3]
```diff
@@ -30,12 +49,73 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+xref:reference:rpk/rpk-topic/rpk-topic.adoc[`rpk topic`]
+
+xref:reference:rpk/rpk-debug/rpk-debug.adoc[`rpk debug`]
+
+xref:reference:rpk/rpk-group/rpk-group.adoc[`rpk group`]
+
+xref:reference:rpk/rpk-redpanda/rpk-redpanda.adoc[`rpk redpanda`]
+
+
+| Security flags got reworked into the global `-X` flag. See xref:reference:rpk/rpk-x-options.adoc[`rpk -X`] for more info.
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| Security flags have been moved into the global `-X` flag. See xref:reference:rpk/rpk-x-options.adoc[`rpk -X`] for details.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 4]
```diff
@@ -30,12 +49,73 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+
+| `--client-id`
+
+`--client-secret`
+
+|23.2.1
+| xref:reference:rpk/rpk-cloud/rpk-cloud-byoc-install.adoc[`rpk cloud byoc install`]
+
+| Use X instead.
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| Use `-X` instead.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 5]
```diff
@@ -30,12 +49,73 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+| `--datasource`
+
+`--job-name`
+
+`--metrics-endpoint`
+
+|23.2.1
+| xref:reference:rpk/rpk-generate/rpk-generate-grafana-dashboard.adoc[`rpk generate grafana dashboard`]
+
+| Use X instead.
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| Use `-X` instead.
```

**Review by Feediver1 — APPROVED** [review]
Please see updates. Thx

File: modules/upgrade/pages/deprecated/index.adoc (hunk 4 above)
Comment (Deflaimun) [reply modules/upgrade/pages/deprecated/index.adoc]: This is just a placeholder for the actual value.

File: modules/upgrade/pages/deprecated/index.adoc (hunk 5 above)
Comment (Deflaimun) [reply modules/upgrade/pages/deprecated/index.adoc]: This is just a placeholder for the actual value.

File: modules/upgrade/pages/deprecated/index.adoc [hunk 6]
```diff
@@ -30,12 +49,74 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+xref:reference:rpk/rpk-topic/rpk-topic.adoc[`rpk topic`]
+
+xref:reference:rpk/rpk-debug/rpk-debug.adoc[`rpk debug`]
+
+xref:reference:rpk/rpk-group/rpk-group.adoc[`rpk group`]
+
+xref:reference:rpk/rpk-redpanda/rpk-redpanda.adoc[`rpk redpanda`]
+
+
+| Security flags have been moved into the global `-X` flag. See xref:reference:rpk/rpk-x-options.adoc[`rpk -X`] for details.
```
Comment (r-vasquez) [inline modules/upgrade/pages/deprecated/index.adoc]: worth mentioning that they still work, they just don't appear in the commands help text anymore.

File: modules/upgrade/pages/deprecated/index.adoc (hunk 4 above)
Comment (r-vasquez) [reply modules/upgrade/pages/deprecated/index.adoc]: The values are: 
```
cloud.client_id=somestring
cloud.client_secret=somelongerstring
```

Also worth mentioning that the flags still work.

btw, can you confirm that these flags were present here before? By looking at the code I don't see any particular change in that regard.

File: modules/upgrade/pages/deprecated/index.adoc (hunk 5 above)
Comment (r-vasquez) [reply modules/upgrade/pages/deprecated/index.adoc]: Use `--dashboard` instead.

Also, old flags still works  :+1:

File: modules/upgrade/pages/deprecated/index.adoc (hunk 6 above)
Comment (Deflaimun) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| Security flags have been moved into the global `-X` flag. While the old ones still work, its recommended that you use the new format. See xref:reference:rpk/rpk-x-options.adoc[`rpk -X`] for details. 
```

File: modules/upgrade/pages/deprecated/index.adoc (hunk 4 above)
Comment (Deflaimun) [reply modules/upgrade/pages/deprecated/index.adoc]: @r-vasquez from what I got, the flags were removed from `rpk cloud byoc install`. They're still present in `rpk cloud byoc`

Comment (r-vasquez) [reply modules/upgrade/pages/deprecated/index.adoc]: Ok, let's just specify that 

```
-X cloud.client_id=somestring 
-X cloud.client_secret=somelongerstring
```

Can be used and we are good to go :+1: thanks

**Review by r-vasquez — APPROVED** [review]
I'm +1 with everything. However, I would not rip the byoc page off.

`rpk cloud byoc` command is a unique command because it downloads the `byoc` plugin from cloud and executes many operations:

```
rpk cloud byoc aws apply
```
For example, creates a BYOC cluster in AWS, but this is not part of rpk, instead, this is something rpk downloads and executes (that's why you don't see it in the help docs)

File: modules/reference/pages/rpk/index.adoc [hunk 7]
```diff
@@ -10,3 +10,5 @@ You can also get descriptions for each flag by running `rpk <command> --help` in
 All `rpk` commands feature autocompletion. To use the feature, press tab. 
 
 For more information on autocompletion, see xref:./rpk-generate/rpk-generate-shell-completion.adoc[`rpk generate shell-completion`]. 
+
+For more information on deprecated `rpk` commands and flags, see xref:upgrade:deprecated/index.adoc#rpk-commands[Deprecated features - rpk commands].
```
Comment (micheleRP) [inline modules/reference/pages/rpk/index.adoc]:
```suggestion
For more information on deprecated `rpk` commands and flags, see xref:upgrade:deprecated/index.adoc[Deprecated features - rpk commands].
```
I was initially confused with the direct link into the rpk commands section, so suggest linking to the top of the page.

Comment (micheleRP) [reply modules/reference/pages/rpk/index.adoc]: Also, there's a back tick typo in the description for `rpk registry` on this index page that you could clean up in this PR. I think it just needs a space so the `rpk` renders properly. And there's also an instance of rpk in the intro that should be `rpk`.

Comment (micheleRP) [reply modules/reference/pages/rpk/index.adoc]: And I think the intro is a little hard to read, with so many short sentences each in a separate paragraph. Maybe if you put this line into a note, it'll break up the intro content and look easier to digest: 
All `rpk` commands feature autocompletion. To use the feature, press tab.

File: modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc.adoc [hunk 8]
```diff
@@ -2,14 +2,13 @@
...
-then proceeds to provision further infrastructure and eventually, a full
-Redpanda cluster.
-
-The BYOC command runs Terraform to create and start the agent. You first need
-a redpanda-id (or cluster ID); this is used to get the details of how your
-agent should be provisioned. You can create a BYOC cluster in our cloud UI
-and then come back to this command to complete the process.
+include::reference:partial$rpk-cloud-byoc-concept.adoc[]
+
+For example, you can create a BYOC cluster in AWS by executing this command:
```
Comment (micheleRP) [inline modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc.adoc]:
```suggestion
For example, to create a BYOC cluster in AWS, run: 
```

File: modules/reference/partials/rpk-cloud-byoc-concept.adoc [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Redpanda installs an agent service in your owned cluster for BYOC. The agent
+then proceeds to provision further infrastructure and eventually, a full
+Redpanda cluster. The command downloads the `byoc` plugin from Redpanda Cloud and can execute many operations.
```
Comment (micheleRP) [inline modules/reference/partials/rpk-cloud-byoc-concept.adoc]:
```suggestion
Redpanda installs an agent service in your BYOC cluster. The agent
then provisions infrastructure and, eventually, a full
Redpanda cluster. The command downloads the `byoc` plugin from Redpanda Cloud and can execute many operations.
```

File: modules/reference/partials/rpk-cloud-byoc-concept.adoc [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
+The BYOC command runs Terraform to create and start the agent. You first need
+a `redpanda-id`` (or cluster ID); this is used to get the details of how your
+agent should be provisioned. You can create a BYOC cluster in our cloud UI
+and then come back to this command to complete the process.
```
Comment (micheleRP) [inline modules/reference/partials/rpk-cloud-byoc-concept.adoc]:
```suggestion
The BYOC command runs Terraform to create and start the agent. You first need
a `redpanda-id` (or cluster ID); this is used to get the details of how your
agent should be provisioned. You can create a BYOC cluster in the Redpanda Cloud UI
and then come back to this command to complete the process.
```

File: modules/reference/partials/rpk-cloud-byoc-concept.adoc (hunk 9 above)
Comment (micheleRP) [reply modules/reference/partials/rpk-cloud-byoc-concept.adoc]: The first line on this page should use Redpanda Cloud; i.e., 
Manage a Redpanda Cloud BYOC agent.

File: modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc-install.adoc [hunk 11]
```diff
@@ -2,10 +2,21 @@
 
 Install the BYOC plugin.
 
+include::reference:partial$rpk-cloud-byoc-concept.adoc[]
+
 This command downloads the BYOC managed plugin if necessary. The plugin is
 installed by default if you try to run a non-install command, but this command
 exists if you want to download the plugin ahead of time.
```
Comment (micheleRP) [inline modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc-install.adoc]:
```suggestion
This command downloads the BYOC managed plugin, if necessary. The plugin is
installed by default if you run a non-install command. This command
exists if you want to download the plugin ahead of time.
```

File: modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc-uninstall.adoc [hunk 12]
```diff
@@ -2,6 +2,8 @@
 
 Uninstall the BYOC plugin.
 
+include::reference:partial$rpk-cloud-byoc-concept.adoc[]
+
 This command deletes your locally downloaded BYOC managed plugin if it exists.
 Often, you only need to download the plugin to create your cluster once, and
 then you never need the plugin again. You can uninstall to save a small bit of
```
Comment (micheleRP) [inline modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc-uninstall.adoc]:
```suggestion
This command deletes your locally-downloaded BYOC managed plugin, if it exists.
You generally only need to download the plugin one time to create your cluster, and
then you never need the plugin again. You can uninstall it to save a small bit of
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 13]
```diff
@@ -14,6 +14,25 @@ This index helps you to identify deprecated features in Redpanda releases and pl
...
+
+| xref:23.2@reference:redpanda-operator/index.adoc[Cluster and Console custom resources]
+| 23.2.1
+| Use the xref:23.2@upgrade:deprecated/cluster-resource.adoc[Redpanda resource] instead.
+
+|===
+
+## rpk commands
+
+All rpk commands that have been modified or have had flag updates.
```
Comment (micheleRP) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
The following `rpk` commands have been modified or have had flag updates.
```

**Review by micheleRP — APPROVED** [review]
left suggested copyedits: let me know if you have questions!

File: modules/reference/pages/rpk/index.adoc (hunk 7 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/index.adoc]: Yeah I hated how this intro came to be. We kept adding more stuff into it and now it's not good

File: modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc.adoc (hunk 8 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-cloud/rpk-cloud-byoc.adoc]: removed example. Will be added a new a ticket

**Pull request merged**